### PR TITLE
feat: improve sandbox routing

### PR DESCRIPTION
**Pull request by agraebe**
fixing #433

### CODE REVIEW

**Comment by friedger** [comment]
fixes also https://github.com/blockstack/explorer/issues/234 :-) Thank you!

**Review by aulneau — APPROVED** [review]
Really great work @agraebe ! Just a few comments

File: src/sandbox/components/clarity-editor/define-theme.ts [hunk 1]
```diff
@@ -207,7 +207,7 @@ export function defineTheme(monaco: any) {
       { fontStyle: 'italic', foreground: '#d19a66', token: 'parameter.variable' },
       {
         foreground: '#abb2bf',
-        token: 'entity.name.function-call',
+        token: 'entity.name.contract-call',
```
[the remark this answers is not in the document]
Comment (agraebe) [reply src/sandbox/components/clarity-editor/define-theme.ts:210]: great call, a mistake. will remove!

File: src/sandbox/components/screens/call-functions/view.tsx [hunk 2]
```diff
@@ -101,10 +101,17 @@ const DefaultContracts: React.FC = () => {
...
 };
 
-export const FunctionCallView: React.FC = () => {
+export const FunctionCallView: React.FC<{ sender: string; contract: string }> = props => {
+  const { sender, contract } = props;
   const [view, setView] = useRecoilState(functionCallViewState);
   const setQuery = useSetRecoilState(contractSearchQueryState);
 
+  // if sender or contract provided, set query
+  if (sender.length > 0 || contract.length > 0) {
```
[the remark this answers is not in the document]
Comment (agraebe) [reply src/sandbox/components/screens/call-functions/view.tsx:110]: thx!

File: src/sandbox/components/page-content.tsx [hunk 3]
```diff
@@ -14,22 +13,15 @@ import { TransactionsPanel } from '@sandbox/components/transactions-panel';
 import { rightPanelState } from '@sandbox/store/views';
 import { useConnect } from '@sandbox/hooks/use-connect';
 import { TokenTransferView } from '@sandbox/components/views/token-transfer';
+import SandboxData from '../../pages/sandbox/[...param]';
```
[the remark this answers is not in the document]
Comment (agraebe) [reply src/sandbox/components/page-content.tsx:16]: this import is actually unused. I'll remove it

**Pull request merged**